### PR TITLE
fix(auth): catch Clerk token throws and enforce banned flag (PF-686)

### PR DESCRIPTION
**Pull request by Tristan578**
## Summary

Three real security bugs found while writing auth edge-case tests (Boy Scout rule — all fixed in this PR):

1. **`auth()` throws propagate as 500s.** Expired/malformed Clerk tokens caused the auth helpers to throw uncaught, turning a 401 scenario into a 500. Now wrapped in try/catch in both `authenticateRequest` and `authenticateClerkSession`.
2. **`users.banned` column was never checked.** The schema has `banned: integer('banned').notNull().default(0)` at `db/schema.ts:66`, but neither auth helper ever read it. Banned users could access every authenticated route. Both helpers now return 403 `ACCOUNT_BANNED` when `banned > 0`.
3. **Clerk 404 timing side-channel.** `attemptSyncWithRetry` waited 500ms and retried on Clerk 404 (user deleted), leaking timing information that distinguished a deleted Clerk user from a transient DB failure. Now detected via `isClerk404()` helper (status field or `/\b404\b|not[_ ]?found/i` in message) and early-returned without retry.

Also adds a **schema-locking regression test** that fails if a future refactor adds `banned` to `syncUserFromClerk`'s `onConflictDoUpdate.set()` block — a partial set on `ON CONFLICT DO UPDATE` preserves unset columns, so omitting `banned` is load-bearing security behavior and easy to break by accident.

Closes #7013 (PF-686)

## Files changed

- `web/src/lib/auth/api-auth.ts` — try/catch around `auth()`, banned check in both helpers, `isClerk404()` helper, early return on 404 in retry path
- `web/src/lib/auth/__tests__/api-auth.test.ts` — 48 tests total, up from ~20. Fixes two lying tests that claimed "returns 401 when auth() throws" but asserted `rejects.toThrow`
- `.changeset/pf-686-auth-edge-cases.md` — patch bump

## Test plan

- [x] `npx vitest run src/lib/auth/__tests__/api-auth.test.ts` — 48/48 pass
- [x] `npx eslint --max-warnings 0 src/lib/auth/` — clean
- [x] `npx tsc --noEmit` — clean
- [ ] Reviewer confirms banned flow works end-to-end (loaded-from-DB and freshly-synced both hit the single banned check)
- [ ] Reviewer confirms `isClerk404()` covers the Clerk SDK's actual error shape

### CODE REVIEW

**Comment by Tristan578** [comment]
## Review board round 1 addressed in 24115adf

**Architect FAIL** (Clerk 404 returns 503):
Fixed. `attemptSyncWithRetry` now returns a discriminated `SyncResult` (`{ kind: 'ok' | 'deleted' | 'degraded' }`) and `authenticateRequest` maps `deleted` → 401 `STALE_SESSION`. See `api-auth.ts:60-94` and the test at `api-auth.test.ts:340-370`.

**Architect IMPORTANT** (`isClerk404` regex too broad):
Fixed in two ways: (1) split `attemptSyncWithRetry`'s try/catch so `isClerk404` is ONLY called on errors from `client.users.getUser()` — DB errors from `syncUserFromClerk` cannot reach it. (2) Narrowed regex from `/not[_ ]?found/i` to `/user not found/i` as defense-in-depth. Regression test added at `api-auth.test.ts:400-430` asserting a DB "record not found" error still retries and 503s (never STALE_SESSION).

**DX FAIL** (401s indistinguishable):
Fixed. All 401 responses now include a `reason` sub-code: `AUTH_PROVIDER_ERROR`, `NO_SESSION`, `CLERK_NOT_CONFIGURED`, or `STALE_SESSION`. Extracted `unauthorized(reason)` and `bannedResponse()` helpers so both auth helpers stay consistent.

**DX IMPORTANT** (schema-lock failure message):
Both assertion messages expanded to explain WHY the `onConflictDoUpdate.set` block's absence / `banned`'s presence is load-bearing security.

**UX FAIL** (banned users no appeal path):
Fixed. `ACCOUNT_BANNED` message now reads "This account has been suspended. If you believe this is a mistake, contact support@spawnforge.ai to appeal." Test asserts the support email is present.

**UX confirmation** (no regression on expired-token 500→401): Reviewer explicitly confirmed this is a strict improvement.

50/50 tests pass. Lint + tsc clean.

Addressed in 24115adf

File: web/src/lib/auth/api-auth.ts [hunk 1]
```diff
@@ -104,19 +188,26 @@ export async function authenticateClerkSession(): Promise<
...
+    return unauthorized('NO_SESSION');
+  }
+
+  // Lightweight banned check: this helper skips the full user sync,
+  // but we must still reject banned users. If the user row is missing
+  // (not yet synced), fall through in the caller's degraded-mode path.
+  const user = await getUserByClerkId(clerkId);
+  if (user && user.banned > 0) {
+    return bannedResponse();
   }
```
[the remark this answers is not in the document]
Comment (Tristan578) [reply web/src/lib/auth/api-auth.ts]: Fixed in d54c0d2a2d67008330cd344ad4754857e3ec0111.

The Sentry finding is valid. `getUserByClerkId` in `authenticateClerkSession` was not wrapped in try/catch, meaning a DB outage would propagate as a 500 error from any route using this helper — directly contradicting the function's documented graceful-degradation contract.

**Fix:** Wrapped the `getUserByClerkId` call and the banned check in a try/catch block. On DB failure the banned check is silently skipped and the caller's degraded-mode path handles the outage (e.g. returning an empty list), which preserves the function's existing contract for routes that can operate without a DB user record.

Also added a regression test (`degrades gracefully when the DB is unavailable during the banned check`) asserting that a `queryWithResilience` throw does not propagate and that the function returns `ok: true` with the correct `clerkId`.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: web/src/lib/auth/api-auth.ts [hunk 2]
```diff
@@ -32,67 +32,151 @@ export async function authenticateRequest(): Promise<
...
   if (!clerkId) {
-    return {
-      ok: false,
-      response: NextResponse.json({ error: 'Unauthorized' }, { status: 401 }),
-    };
+    return unauthorized('NO_SESSION');
   }
 
-  const user = await getUserByClerkId(clerkId);
+  let user = await getUserByClerkId(clerkId);
```
[the remark this answers is not in the document]
Comment (Tristan578) [reply web/src/lib/auth/api-auth.ts]: Fixed in 3621d9f7. Wrapped the `getUserByClerkId` call in `authenticateRequest` (api-auth.ts:54) in a try/catch that returns the same 503 `SERVICE_DEGRADED` payload used by the sync-failure path. A DB-layer throw no longer propagates as a 500 and can no longer leak past into the sync retry flow (regression test in api-auth.test.ts asserts status=503, body.error=SERVICE_DEGRADED, and that `syncUserFromClerk` is never called).

File: web/src/app/api/admin/economics/config/route.ts [hunk 3]
```diff
@@ -18,25 +44,10 @@ export async function PUT(request: NextRequest) {
...
     if (body.type === 'token_config') {
-      if (typeof body.tokenCost !== 'number' || !Number.isFinite(body.tokenCost) || body.tokenCost < 0) {
-        return NextResponse.json({ error: 'tokenCost must be a non-negative finite number' }, { status: 400 });
-      }
-      if (typeof body.estimatedCostCents !== 'number' || !Number.isFinite(body.estimatedCostCents) || body.estimatedCostCents < 0) {
-        return NextResponse.json({ error: 'estimatedCostCents must be a non-negative finite number' }, { status: 400 });
-      }
       await queryWithResilience(() =>
         getDb().update(tokenConfig)
           .set({
```
[the remark this answers is not in the document]
Comment (Tristan578) [reply web/src/app/api/admin/economics/config/route.ts]: Fixed in 3621d9f7. The handler now builds the update payload conditionally — `active` is only included when `body.active !== undefined`. Omitting the optional field is now a true no-op on that column, matching the schema contract. Two regression tests in route.test.ts: one asserts the `.set()` arg has no `active` key when omitted, the other asserts `active: 0` when explicitly set to `false`.
